### ISSUE TITLE
Invalid code logic tripping up some compilers

### ISSUE THREAD

**Issue by gold2718** [ESCOMP/PUMAS#8]
There is logic in PUMAS which  incorrect from a Fortran standard point of view. An example is line 2099 from micro_mg3_0.F90:
```
      if (lamr(i,k) > qsmall .and. 1._r8/lamr(i,k) < Dcs) then
```
The Fortran standard allows the compiler to evaluate `1._r8/lamr(i,k)` even if `lamr(i,k) <= qsmall` and some compilers are doing that (e.g., gnu 9.3.0). Code like this (probably found by looking for instances of `.and.` and `.or.`) needs to be rewritten as:
```
      if (lamr(i,k) > qsmall) then
         if (1._r8/lamr(i,k) < Dcs) then
```

**Comment by Katetc** [comment]
Hi @gold2718 , do you have a test I can run to make sure that I fix all of these issues? How did you trigger the problem? I just want to make sure I fix them all. Thanks!

**Comment by gold2718** [comment]
@Katetc, sorry for the delay, my notifications got turned off. I did not have specific tests but running the aux_cam tests always triggered them.

**Comment by andrewgettelman** [comment]
@Katetc is this complete and can we close it? Or do we still have to make further commits here?

**Comment by Katetc** [comment]
Fixed in commit 65804082, pumas_cam-release_v1.13
